mesonbuild/compilers.py

```diff
@@ -2263,6 +2267,34 @@ def get_gcc_soname_args(gcc_type, prefix, shlib_name, suffix, path, soversion, i
     else:
         raise RuntimeError('Not implemented yet.')
 
+def get_compiler_is_linuxlike(compiler):
+    if (getattr(compiler, 'gcc_type', None) == GCC_STANDARD) or \
+       (getattr(compiler, 'clang_type', None) == CLANG_STANDARD) or \
+       (getattr(compiler, 'icc_type', None) == ICC_STANDARD):
+        return True
+    return False
+
+def get_largefile_args(compiler):
+    '''
+    Enable transparent large-file-support for 32-bit UNIX systems
+    '''
+    if get_compiler_is_linuxlike(compiler):
+        # Enable large-file support unconditionally on all platforms other
+        # than macOS and Windows. macOS is now 64-bit-only so it doesn't
+        # need anything special, and Windows doesn't have automatic LFS.
+        # You must use the 64-bit counterparts explicitly.
+        # glibc, musl, and uclibc, and all BSD libcs support this. On Android,
+        # support for transparent LFS is available depending on the version of
+        # Bionic: https://github.com/android/platform_bionic#32-bit-abi-bugs
+        # https://code.google.com/p/android/issues/detail?id=64613
+        #
+        # If this breaks your code, fix it! It's been 20+ years!
+        return ['-D_FILE_OFFSET_BITS=64']
```

Comment:
Support for 64-bit operations on fts functions was added only a few years ago in glibc [through this commit](https://sourceware.org/git/?p=glibc.git;a=commit;h=8b7b7f75d91f7bac323dd6a370aeb3e9c5c4a7d5) and there are systems that still ship older versions glibc (for e.g. opensuse). Unconditionally enabling this macro on all systems will cause compilation failures. There should be a feature test before this feature macro is enabled.